src/rgw/rgw_http_client.cc

```diff
@@ -253,7 +382,8 @@ int RGWHTTPClient::process(const char *method, const char *url)
   last_method = (method ? method : "");
   last_url = (url ? url : "");
 
-  curl_handle = curl_easy_init();
+  auto ca = handles->get_curl_handle();
```

Comment:
it looks like there are cases where `get_curl_handle()` could return null, so we should check for that here and return a failure